etcdctl/ctlv3/command/printer_simple.go

```diff
@@ -141,7 +141,11 @@ func (s *simplePrinter) RoleGet(role string, r v3.AuthRoleGetResponse) {
 	printRange := func(perm *v3.Permission) {
 		sKey := string(perm.Key)
 		sRangeEnd := string(perm.RangeEnd)
-		fmt.Printf("\t[%s, %s)", sKey, sRangeEnd)
+		if strings.Compare(sRangeEnd, "\xff") != 0 {
```

Comment:
Just checking against "\x00" here should be OK, no need for the adt/affine stuff. RangeEnd == "\x00" meaning >= is already part of the RPC wire protocol.